### ISSUE TITLE
is there a way to be notified about underlying disconnects/reconnects from the broker? 

### ISSUE THREAD

**Issue by rotemtam** [confluentinc/confluent-kafka-go#220]
A question,

if my broker goes offline i see in the logs for example
```
%3|1534163670.841|FAIL|rdkafka#consumer-1| [thrd:ip-192-168-10-207.us-west-1.compute.internal:9092/0]: ip-192-168-10-207.us-west-1.compute.internal:9092/0: Receive failed: Connection reset by peer
%3|1534163670.841|ERROR|rdkafka#consumer-1| [thrd:ip-192-168-10-207.us-west-1.compute.internal:9092/0]: ip-192-168-10-207.us-west-1.compute.internal:9092/0: Receive failed: Connection reset by peer
%3|1534163670.841|ERROR|rdkafka#consumer-1| [thrd:ip-192-168-10-207.us-west-1.compute.internal:9092/0]: 2/2 brokers are down
```

is there a way to programmatically be notified when this happens?

**Comment by edenhill** [comment]
Not at this time, there are plans to provide broker, topic, partition and leader events, but nothing concrete yet.

You can use OnStatistics in the meantime and monitor the broker.state:
https://github.com/edenhill/librdkafka/blob/master/STATISTICS.md

**Comment by rotemtam** [comment]
thanks for the quick reply!
correct me if i'm wrong but this forces usage of retrieving messages using consumer.Events(), correct?

i cant use the function API (consumer.ReadMessage) and consumer.Events() together as far as i can tell

**Comment by onorua** [comment]
I believe it is implemented in v0.11.6:
```
	for e := range p.Producer.Events() {
		switch ev := e.(type) {
		case *kafka.Message:
                        // reply message ....
		case kafka.Error:
			if ev.Code() == kafka.ErrAllBrokersDown {
				// do what is required when brokers are down
			}
		}
```
I believe this issue can be closed

**Comment by dtelyukh** [comment]
I confirm that it fixed in v0.11.6 (I had the same issue).
